sotopia/cli/benchmark/benchmark.py

```diff
@@ -39,86 +44,319 @@
 from pathlib import Path
 from ..app import app
 
+# date and message only
+FORMAT = "%(asctime)s - %(levelname)s - %(name)s - %(message)s"
+
+process = subprocess.Popen(
+    ["git", "rev-parse", "HEAD"], shell=False, stdout=subprocess.PIPE
+)
+git_head_hash = process.communicate()[0].strip()
+
+logging.basicConfig(
+    level=20,
+    format=FORMAT,
+    datefmt="[%X]",
+    handlers=[
+        RichHandler(),
+    ],
+)
+
+default_model_list: list[str] = [
+    "gpt-4o",
+]
+dimension_range_mapping = OrderedDict(
+    {
+        "social_rules": ["SOC", [-10, 0]],
+        "secret": ["SEC", [-10, 0]],
+        "financial_and_material_benefits": ["FIN", [-5, 5]],
+        "relationship": ["REL", [-5, 5]],
+        "knowledge": ["KNO", [0, 10]],
+        "goal": ["GOAL", [0, 10]],
+        "believability": ["BEL", [0, 10]],
+    }
+)
+
+
+def get_avg_reward(
+    episodes: list[EpisodeLog], model_name: str, binary: bool = False
+) -> dict[str, tuple[float, float]]:
+    """
+    return: dictionary of {dimension: (avg_reward, margin_of_error (in 95% confidence interval))}, plus the distinct setting number and episode count (in the same format, but with 0 margin of error)
+    """
+    rewards_dict = defaultdict(
+        list
+    )  # {pk: [rewards]}, {pk}_{i} denotes the i-th agent is the test agent
+    avg_reward_dict = {}
+    avg_margin_dict = {}
+    avg_variance_dict = {}
+    for episode in episodes:
+        assert episode.models is not None, "episode.models should not be None"
+        if episode.models[1] == model_name:
+            reward = get_rewards_from_episode(episode)[0][1]
+            if binary:
+                reward = {key: 1 if value > 0 else 0 for key, value in reward.items()}
+            rewards_dict[f"{episode.environment}_0"].append(reward)
+        else:
+            reward = get_rewards_from_episode(episode)[1][1]
+            if binary:
```

Comment:
I am a bit confused. Are we using this binary indicator to show if there may be some risks? If so the reward should be 1 if there are risks related (i.e. <0 rather than > 0)?

Reply:
AH good catch, let me fix this